internal/controller/openstackclusterstackrelease_controller.go

```diff
@@ -174,8 +176,22 @@ func (r *OpenStackClusterStackReleaseReconciler) Reconcile(ctx context.Context,
 		}
 
 		osnirName := fmt.Sprintf("%s-%s-%s", nameWithoutVersion, openStackNodeImage.CreateOpts.Name, nodeImageVersion)
+		cloudName, err := r.getCloudNameFromSecret(ctx, openstackclusterstackrelease.Namespace, openstackclusterstackrelease.Spec.IdentityRef.Name)
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				conditions.MarkFalse(openstackclusterstackrelease,
+					apiv1alpha1.CloudNameAvailableCondition,
+					apiv1alpha1.SecretNotFoundReason,
+					clusterv1beta1.ConditionSeverityError,
+					err.Error(),
+				)
+				record.Warnf(openstackclusterstackrelease, "SecretNotFound", err.Error())
+				logger.Error(err, "failed to get secret")
+				return ctrl.Result{RequeueAfter: 1 * time.Minute}, nil
+			}
```

Comment:
right now you don't return in case an error that is different than NotFound comes. Is that intentional?

Reply:
Yes, I know that. I just didn't think of another error that could occur, maybe just an error during the parsing of JSON-encoded data in the secret(InvalidUnmarshalError).

Reply:
maybe a temporary Kubernetes API server issue.. Can be anything. I guess you should return an error in case it is an unknown one. At least if you don't explicitly want to ignore it

Reply:
I will modify it so that it also catches the unknown error

Reply:
see b3d97e990e7d5f3c6a576ed1f4dc88feababc096

Reply:
Personally I would say that you don't need to specify any condition and can just return the error. 
AFAIK it can only be a temporary error. 
Therefore, in case you wanna keep the condition, you can probably set its severity on "warning" instead of "error", because the user should not have to do anything.

Reply:
changed the severity to `warning`